### PR TITLE
Enhancements for Browse building floors sample

### PR DESCRIPTION
**Pull request by shubham7109**
## Description
PR to support "vertical order" for the Browse building floors sample. Since buildings can have basements, the starting floor should be set to `verticalOrder == 0` instead of using `currentFloor = 0`.

Changes made:
- Removed hardcoded floors, now we use a spinner adapter to retrieve floor names `floorLevels[position].longName` 
- Added a spinner item layout file
- Added support to use `verticalOrder == 0` as the default floor on app launch, instead of the first floor in `floorManager.levels`

The sample will now be compatible with any floor-aware webmap! 

## Links and Data
Sample Epic: `runtime/common-samples/issues/3465`

## What To Review
-  Review the code to make sure it is easy to follow like other samples on Android
- The sample defaults to use `verticalOrder == 0` and can switch between other floors 

## How to Test
Run the sample on the sample viewer or the repo.

### CODE REVIEW

File: kotlin/browse-building-floors/src/main/java/com/esri/arcgisruntime/sample/browsebuildingfloors/MainActivity.kt [hunk 1]
```diff
@@ -66,62 +65,58 @@ class MainActivity : AppCompatActivity() {
...
+                            // ignore if nothing is selected
+                            override fun onNothingSelected(parentView: AdapterView<*>?) {}
+                        }
+                    // select the ground floor using `verticalOrder`.
+                    // in the case of buildings with basements, they can have negative
+                    // vertical orders; ground floor is expected to be 0.
+                    // you can also use level ID, number or name to locate a floor.
+                    setSelection(floorManager.levels.indexOf(
+                        floorManager.levels.first { it.verticalOrder == 0 }
+                    ))
```
Comment (shubham7109) [inline kotlin/browse-building-floors/src/main/java/com/esri/arcgisruntime/sample/browsebuildingfloors/MainActivity.kt:117]: Cool kotlin-y approach to resolve the default floor to the ground floor

File: kotlin/browse-building-floors/src/main/java/com/esri/arcgisruntime/sample/browsebuildingfloors/MainActivity.kt [hunk 2]
```diff
@@ -66,62 +65,58 @@ class MainActivity : AppCompatActivity() {
...
         map.addDoneLoadingListener {
             if (map.loadStatus == LoadStatus.LOADED && map.floorDefinition != null) {
-
                 // get and load the floor manager
-                val floorManager = map.floorManager
-                floorManager.loadAsync()
-
-                // set initial floor level to currentFloor
+                val floorManager = map.floorManager.apply { loadAsync() }
+                // set initial floor level to ground floor
```
Comment (TADraeseke) [inline kotlin/browse-building-floors/src/main/java/com/esri/arcgisruntime/sample/browsebuildingfloors/MainActivity.kt:70]: I think this was better before. Remember we're trying to use the `apply` block as an extended builder.

someObject(some paramater).apply {
    someOtherParamater = blah
    yetSome Other = blah 2
}

Let's reserve things like .loadAsync() and .cancel() to calls to the parent object, just for readability

File: kotlin/browse-building-floors/src/main/res/layout/browse_floors_spinner_item.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<TextView xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/list_item"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:padding="10dp" />
```
Comment (TADraeseke) [inline kotlin/browse-building-floors/src/main/res/layout/browse_floors_spinner_item.xml:6]: missing new line after

Comment (TADraeseke) [inline kotlin/browse-building-floors/src/main/res/layout/browse_floors_spinner_item.xml:6]: missing new line after

**Review by TADraeseke — APPROVED** [review]
LGTM! Just a few small things @shubham7109!

**Review by hud10837 — CHANGES_REQUESTED** [review]
A few questions/changes :)

File: kotlin/browse-building-floors/src/main/java/com/esri/arcgisruntime/sample/browsebuildingfloors/MainActivity.kt [hunk 4]
```diff
@@ -66,62 +65,58 @@ class MainActivity : AppCompatActivity() {
...
-            android.R.layout.simple_spinner_item
-        ).apply {
-            setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
-        }
-
     }
 
     /**
-     * update the [floorManager] to the currently selected floor
+     * Load and update the [floorManager] to the currently selected floor
```
[the remark this answers is not in the document]
Comment (shubham7109) [reply kotlin/browse-building-floors/src/main/java/com/esri/arcgisruntime/sample/browsebuildingfloors/MainActivity.kt:82]: Updated the function comment

File: kotlin/browse-building-floors/src/main/java/com/esri/arcgisruntime/sample/browsebuildingfloors/MainActivity.kt [hunk 5]
```diff
@@ -66,62 +65,58 @@ class MainActivity : AppCompatActivity() {
...
      * and disable the other floors.
      */
     private fun setFloor(floorManager: FloorManager) {
```
[the remark this answers is not in the document]
Comment (shubham7109) [reply kotlin/browse-building-floors/src/main/java/com/esri/arcgisruntime/sample/browsebuildingfloors/MainActivity.kt:85]: Good point, this was the approach used in another sample. Renamed to `initializeFloorSpinner`

**Comment by shubham7109** [comment]
Thanks for the detailed feedback @hud10837, ready for re-review 👍

**Review by hud10837 — APPROVED** [review]
👍 Looks great to me

**Review by saratchandrakarumuri — APPROVED** [review]
sample is working as expected

**Pull request merged**